central/vuln-mgmt/workload/service/service_impl_postgres_test.go

```diff
@@ -0,0 +1,218 @@
+//go:build sql_integration
+
+package service
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"testing"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/pkg/errors"
+	deploymentDS "github.com/stackrox/rox/central/deployment/datastore"
+	imageDS "github.com/stackrox/rox/central/image/datastore"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/fixtures"
+	"github.com/stackrox/rox/pkg/fixtures/fixtureconsts"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+func TestServicePostgres(t *testing.T) {
+	suite.Run(t, new(servicePostgresTestSuite))
+}
+
+type servicePostgresTestSuite struct {
+	suite.Suite
+
+	ctx         context.Context
+	pool        *pgtest.TestPostgres
+	deployments deploymentDS.DataStore
+	images      imageDS.DataStore
+	service     Service
+}
+
+func (s *servicePostgresTestSuite) SetupTest() {
+	s.ctx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Deployment, resources.Image),
+		),
+	)
+	s.pool = pgtest.ForT(s.T())
+	s.Require().NotNil(s.pool)
+
+	deployments, err := deploymentDS.GetTestPostgresDataStore(s.T(), s.pool)
+	s.Require().NoError(err)
+	s.deployments = deployments
+	s.images = imageDS.GetTestPostgresDataStore(s.T(), s.pool)
+	s.service = New(s.deployments, s.images)
+}
+
+func (s *servicePostgresTestSuite) TearDownTest() {
+	s.pool.Teardown(s.T())
+	s.pool.Close()
+}
+
+// authInterceptor overrides the server context to make sure all calls are authenticated.
+func (s *servicePostgresTestSuite) authInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	return handler(srv, &grpc_middleware.WrappedServerStream{
+		ServerStream:   ss,
+		WrappedContext: s.ctx,
+	})
+}
+
+func (s *servicePostgresTestSuite) createGRPCWorkloadsService(ctx context.Context) (*grpc.ClientConn, func()) {
```

Comment:
Given that we have this construct within the roxctl tests as well as here now (i.e. creating a gRPC server, registering some service with it for testing) might make sense to move it within the gRPC package as a test utility; but more as a follow-up instead of within this PR.